indexer/src/main.rs

```diff
@@ -31,27 +34,49 @@ fn run(home_dir: std::path::PathBuf, config: RunConfigArgs) -> Result<()> {
     };
 
     let system = actix::System::new();
+
+    let registry = Registry::new();
+    let server_handle = system.block_on(async {
```

Comment:
`run_metrics_server` only spawns a separate `tokio` routine. You can do that here without a blocking the system. Just call the `run_metrics_server` here.

Reply:
It looks like the runtime isn't available this way. 
Running the following 
```
    let server_handle = if let Some(metrics_addr) = config.metrics_ip_port_address {
        Some(run_metrics_server(metrics_addr, registry.clone()))
    } else {
        None
    };
```

Results in error:
`there is no reactor running, must be called from the context of a Tokio 1.x runtime
`

Reply:
But to make non-blocking  system.runtime().spawn() can be used

Reply:
Yeah, since you rely on `actix` to create a reactor, and we also have no guarantees that `actix` will always use `tokio`, you should use the `actix`-way to spawn something, or initialize the reactor with `actix::main` attribute.

Reply:
Merged in another PR where this is done